### PR TITLE
Fixes #21088: Reporting issues when using multiple directives of the same technique on a node

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/21088

### CODE REVIEW

File: tree/20_cfe_basics/log_rudder.cf [hunk 1]
```diff
@@ -516,11 +516,13 @@ bundle agent _method_reporting_context_v4(c_name, c_key, report_id)
     "report_data.report_id_r"    string => "${report_id}";
     # Canonified version, to be used in policies (as class prefix)
     "report_data.report_id"      string => canonify("${report_id}");
-    "report_data.method_id"      string => canonify("${report_id}");
+    "report_data.method_id"      string => canonify("${report_id}${report_data.directive_id}");
```
Comment (amousset) [inline tree/20_cfe_basics/log_rudder.cf:519]:
```suggestion
    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");
```

File: tree/20_cfe_basics/log_rudder.cf [hunk 2]
```diff
@@ -516,11 +516,13 @@ bundle agent _method_reporting_context_v4(c_name, c_key, report_id)
     "report_data.report_id_r"    string => "${report_id}";
     # Canonified version, to be used in policies (as class prefix)
     "report_data.report_id"      string => canonify("${report_id}");
-    "report_data.method_id"      string => canonify("${report_id}");
+    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");
     "report_data.identifier"     string => "${report_data.rule_id}@@${report_data.directive_id}@@${report_data.report_id_r}";
 
   methods:
-    "${report_id}" usebundle => rudder_method_id_reset;
```
Comment (VinceMacBuche) [inline tree/20_cfe_basics/log_rudder.cf:523]: I think the promiser should be report_data.method_id or else it won't be rexecuted if it has already been done once for this report_id

File: tree/20_cfe_basics/log_rudder.cf [hunk 3]
```diff
@@ -516,11 +516,13 @@ bundle agent _method_reporting_context_v4(c_name, c_key, report_id)
     "report_data.report_id_r"    string => "${report_id}";
     # Canonified version, to be used in policies (as class prefix)
     "report_data.report_id"      string => canonify("${report_id}");
-    "report_data.method_id"      string => canonify("${report_id}");
+    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");
```
Comment (ncharles) [inline tree/20_cfe_basics/log_rudder.cf:519]: ok, this change the semantic used, and it seems to break package

**Comment by Fdall** [comment]
Closing as it is now in https://github.com/Normation/ncf/pull/1353

**Pull request closed without merging**